contracts/core/MigrationRibbon.sol

```diff
@@ -0,0 +1,169 @@
+
+import {NeuronCollateralVaultInterface} from "../interfaces/NeuronCollateralVaultInterface.sol";
+import {RibbonVaultInterface} from "../interfaces/RibbonVaultInterface.sol";
+
+contract MigrationRibbon is UUPSUpgradeable, OwnableUpgradeable {
+    // ----------------------------------------------------------
+    // ---------------------  STRUCTURES  -----------------------
+    // ----------------------------------------------------------
+
+    struct DepositReceipt {
+        address recipient;
+        uint256 amount;
+    }
+
+    // ----------------------------------------------------------
+    // ----------------------  CONSTANTS  -----------------------
+    // ----------------------------------------------------------
+
+    IERC20 internal constant WETH = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    // ----------------------------------------------------------
+    // -----------------------  STORAGE  ------------------------
+    // ----------------------------------------------------------
+
+    RibbonVaultInterface public ribbonVault;
+
+    IERC20 public ribbonAssetToken;
+
+    NeuronCollateralVaultInterface public neuronCollateralVault;
+
+    mapping(uint16 => DepositReceipt[]) public depositReceipts;
+
+    uint16 public lastRound;
+
+    mapping(uint16 => uint256) internal amounts;
+
+    // ----------------------------------------------------------
+    // -----------------------  EVENTS  -------------------------
+    // ----------------------------------------------------------
+
+    event Deposit(address indexed recipient, uint256 amount);
+
+    event Withdraw(uint16 indexed round);
+
+    // ----------------------------------------------------------
+    // -----------------------  ERRORS  -------------------------
+    // ----------------------------------------------------------
+
+    error NotEnoughFunds();
+
+    error RoundNotCompleted();
+
+    error FundsLastRoundNotWithdrawn();
+
+    // ----------------------------------------------------------
+    // -----------------------  GETTERS  ------------------------
+    // ----------------------------------------------------------
+
+    function getCurrentRound() public view returns (uint16) {
+        return ribbonVault.vaultState().round;
+    }
+
+    // ----------------------------------------------------------
+    // -----------------------  SETTERS  ------------------------
+    // ----------------------------------------------------------
+
+    function setNeuronCollateralVault(address _neuronCollateralVault) external {
+        neuronCollateralVault = NeuronCollateralVaultInterface(_neuronCollateralVault);
+    }
+
+    // ----------------------------------------------------------
+    // ---------------------  INITIALIZE  -----------------------
+    // ----------------------------------------------------------
+
+    function initialize(address _ribbonVault, address _neuronCollateralVault) public initializer {
```

Comment:
Instead of `address _neuronCollateralVault` provide `address[] calldata _neuronCollateralVaults` and add all vauls to mapping